balena/models/device.py

```diff
@@ -1361,3 +1367,158 @@ def get_os_update_status(self, uuid):
             )
         )
+
+    def __is_provisioned_device(self, device):
+        return device['supervisor_version'] and device['last_connectivity_event']
+
+    def __check_local_mode_supported(self, device):
+        if not self.__is_provisioned_device(device):
+            raise exceptions.LocalModeError(Message.DEVICE_NOT_PROVISIONED)
+
+        if semver.compare(self.device_os._DeviceOs__normalize_balena_semver(device['os_version']), LOCAL_MODE_MIN_OS_VERSION) < 0:
+            raise exceptions.LocalModeError(Message.DEVICE_OS_NOT_SUPPORT_LOCAL_MODE)
+
+        if semver.compare(self.device_os._DeviceOs__normalize_balena_semver(device['supervisor_version']), LOCAL_MODE_MIN_SUPERVISOR_VERSION) < 0:
+            raise exceptions.LocalModeError(Message.DEVICE_SUPERVISOR_NOT_SUPPORT_LOCAL_MODE)
+
+        if device['os_variant'] != 'dev':
+            raise exceptions.LocalModeError(Message.DEVICE_OS_TYPE_NOT_SUPPORT_LOCAL_MODE)
+
+    def __upsert_device_local_mode(self, device, value):
+        try:
+            data = {
+                'device': device['id'],
+                'application': device['belongs_to__application']['__id'],
+                'name': LOCAL_MODE_ENV_VAR,
+                'value': value
+            }
+
+            return self.base_request.request(
+                'device_config_variable', 'POST', data=data,
+                endpoint=self.settings.get('pine_endpoint')
+            )
+        except exceptions.RequestError as e:
+            if 'Unique key constraint violated' in e.message or 'must be unique' in e.message:
+                params = {
+                    'filters': {
+                        'device': device['id'],
+                        'name': LOCAL_MODE_ENV_VAR
+                    }
+                }
+
+                data = {
+                    'value': value
+                }
+
+                return self.base_request.request(
+                    'device_config_variable', 'PATCH', params=params, data=data,
+                    endpoint=self.settings.get('pine_endpoint')
+                )
+            raise e
+
+    def is_in_local_mode(self, uuid):
+        """
+        Check if local mode is enabled on the device.
+
+        Args:
+            uuid (str): device uuid.
+
+        Returns:
+            bool: True if local mode enabled, otherwise False.
+
+        Examples:
+            >>> balena.models.device.is_in_local_mode('b6070f4fea5edf808b576123157fe5ec')
+            True
+
+        """
+
+        device = self.get(uuid)
+
+        params = {
+            'filters': {
+                'device': device['id'],
+                'name': LOCAL_MODE_ENV_VAR
+            }
+        }
+
+        response = self.base_request.request(
+            'device_config_variable', 'GET', params=params,
+            endpoint=self.settings.get('pine_endpoint')
+        )['d']
+
+        if len(response) == 0:
+            return False
+        if response[0]['value'] != '1':
```

Comment:
Optional-NIT: Imo a `== 1` `return True` check is more verbose and follows more the 
> local mode is enabled iff the value equals `1`

PS: I'm not saying it's wrong and I'm fine with this as is.

Reply:
Updated!